### ISSUE TITLE
Only state is restored, not "set" attributes

### ISSUE THREAD

**Issue by NateEaton** [snarky-snark/home-assistant-variables#116]
My use case involves having a list of attributes that are set in an automation with each being the name of a battery sensor and the date it was changed. The automation also sets the value of the variable to the latest sensor/date set as an attribute. 

However, after a restart of HA, the state is restored but the only attribute restored is the one defined in the initial configuration (Friendly Name). 

I am not sure whether this is a bug or a feature request as the documentation only references state, not attributes, but it seems reasonable to expect that both would be restored.

**Comment by estegewr** [comment]
Up to version 0.15.0, I was able to restore attributes by initializing the attributes in the configuration file to an empty string. I think that this was more of a side effect and not a planned behavior. Since version 0.15.1 this is no longer possible.
Therefore I would like to join the feature request. Restoring attributes in the same way as status values would be very helpful. if it simplifies the implementation, I would also name the attributes to be restored in the configuration.

**Comment by amandatennent** [comment]
I came across this issue as well, my work around (in case anyone else finds it helpful) is to create a script and have it run on home assistant start up

**Comment by estegewr** [comment]
Where do you get the values to restore them in the init script, @amandatennent?

**Comment by amandatennent** [comment]
I guess my use case is different, the attributes on my variables are static and aren't modified after they're created. I mis-read  the initial issue.

**Comment by nohumaninput** [comment]
Thank you for the great component.
I have same issue, I'm storing user inputs (via https://github.com/faeibson/lovelace-multiline-text-input-card) in an attribute. After restart of HA the attributes get not restored.
Anyone aware of a workaround?

**Comment by nohumaninput** [comment]
I'm now using an automation to store the value to mqtt. Another automation restoring the value from mqtt to attribute after restart. First tests seem successful.

**Comment by kernel610** [comment]
> Thank you for the great component. I have same issue, I'm storing user inputs (via https://github.com/faeibson/lovelace-multiline-text-input-card) in an attribute. After restart of HA the attributes get not restored. Anyone aware of a workaround?

My exact use case as well.

@nohumaninput To make sure I understand your flow correctly, you are triggering a MQTT message that gets retained and then another automation that runs upon restart, pulls from that retained MQTT message to update the variable attribute? What are you using as the trigger to push the first MQTT message because the state shouldn't change if you are updating an existing attribute.

**Comment by nohumaninput** [comment]
> > Thank you for the great component. I have same issue, I'm storing user inputs (via https://github.com/faeibson/lovelace-multiline-text-input-card) in an attribute. After restart of HA the attributes get not restored. Anyone aware of a workaround?
> 
> My exact use case as well.
> 
> [@nohumaninput](https://github.com/nohumaninput) To make sure I understand your flow correctly, you are triggering a MQTT message that gets retained and then another automation that runs upon restart, pulls from that retained MQTT message to update the variable attribute? What are you using as the trigger to push the first MQTT message because the state shouldn't change if you are updating an existing attribute.

Yes exactly. Took me quite a while to figure this out. :) I just added a condition to both automations to check if it exists or not. Don't know if this is the best approach, but it works for me.

Push variable to mqtt:

- id: _default_set_rest_call_01_mqtt
  alias: _default_set_rest_call_01_mqtt
  trigger:
    - platform: state
      entity_id: var.rest_calls
      attribute: rest_call_01
  condition: '{{ state_attr("var.rest_calls","rest_call_01") != None }}'
  action:
    service: mqtt.publish
    data_template:
      topic: TOPIC
      retain: true
      qos: 2
      payload: '{{ state_attr("var.rest_calls","rest_call_01") }}'


Pull payload after start:

- id: _default_set_rest_call_01_start
  alias: _default_set_rest_call_01_start
  trigger:
    - platform: mqtt
      topic: TOPIC
  condition: '{{ state_attr("var.rest_calls","rest_call_01") == None }}'
  action:
    - service: var.set
      target:
        entity_id: var.rest_calls
      data:
        attributes:
          rest_call_01: '{{ trigger.payload }}'
